### PR TITLE
Add global user selectable list

### PR DESCRIPTION
**Pull request by agamm**
- Add a user-selectable list of last used comebacks.
`comeback -l`
comebackl
Note: need to update README.

### CODE REVIEW

**Comment by agamm** [comment]
fixes https://github.com/agamm/comeback/issues/17

File: comeback/main.py [hunk 1]
```diff
@@ -111,9 +111,12 @@ def get_config_path() -> pathlib.Path:
     return last_comeback_used
 
 
-def load_config() -> None:
+def load_config(config_path = None) -> None:
```
Comment (yammesicka) [inline comeback/main.py]: `config_path: Optional[pathlib.Path] = None`

File: comeback/main.py [hunk 2]
```diff
@@ -141,12 +144,37 @@ def main() -> None:
     load_config()
 
 
+def list_last_used() -> Tuple[List[Dict], str]:
```
Comment (yammesicka) [inline comeback/main.py]: `Dict[str, Any]`

File: comeback/main.py [hunk 3]
```diff
@@ -141,12 +144,37 @@ def main() -> None:
...
+    last_used = []
+    for last_used_path in config.get_recent_comebacks(5):
+        last_used.append({'path': last_used_path[0],
```
Comment (yammesicka) [inline comeback/main.py]: I personally like less line breaks:

```python
        path, last_used = last_used_path
        last_used.append({'path': path, 'last_used': last_used)
```

File: comeback/main.py [hunk 4]
```diff
@@ -141,12 +144,37 @@ def main() -> None:
     load_config()
 
 
+def list_last_used() -> Tuple[List[Dict], str]:
+    last_used = []
```
Comment (yammesicka) [inline comeback/main.py]: I'd move the creation of the dictionary to another function. 
Easier to test, might be used in the future in other functions.

File: comeback/main.py [hunk 5]
```diff
@@ -141,12 +144,37 @@ def main() -> None:
...
+    for last_used_path in config.get_recent_comebacks(5):
+        last_used.append({'path': last_used_path[0],
+                          'last_used': last_used_path[1]})
+
+    sorted_last_used = sorted(last_used, key=lambda k: k['last_used'],
+                              reverse=True)
+
+    return sorted_last_used, "\n".join(f'{i} - {c["path"]}' for i, c in
```
Comment (yammesicka) [inline comeback/main.py]: A bit messy

File: comeback/main.py [hunk 6]
```diff
@@ -141,12 +144,37 @@ def main() -> None:
...
+
 @click.group(invoke_without_command=True)
 @click.pass_context
 @click.option('-i', '--init', is_flag=True, default=False,
               help='Generate a blank .comeback configuration file.')
 @click.option('-v', '--verbose', is_flag=True, help='Show more output.')
-def cli(ctx: click.Context, init: bool, verbose: bool) -> None:
+@click.option('-l', '--last_used', is_flag=True, help='Show recently used' +
+                                                      '.comeback recipes.')
+def cli(ctx: click.Context, init: bool, verbose: bool, last_used: bool) -> None:
```
Comment (yammesicka) [inline comeback/main.py]: \> 79

Comment (agamm) [reply comeback/main.py]: image

Comment (agamm) [reply comeback/main.py]: Did it, happy?

File: comeback/main.py [hunk 7]
```diff
@@ -141,12 +144,44 @@ def main() -> None:
...
+        last_used_list.append({'path': path, 'last_used': last_used})
+
+    return sorted(last_used_list, key=lambda k: k['last_used'],
+                  reverse=True)
+
+
+def list_last_used() -> Tuple[List[Dict[str, Any]], str]:
+    sorted_last_used = get_last_used()
+    sorted_last_str = ""
+    for index, recpie in enumerate(sorted_last_used):
```
Comment (yammesicka) [inline comeback/main.py]: `recipe` ^^"

**Review by yammesicka — APPROVED** [review]

**Pull request merged**